### PR TITLE
Migrate gradle to kotlin

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
# PR Title
### Checklist

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines](/CONTRIBUTING.md).
- [x] I agree with the [license](/LICENSE).

### Changes
- [x] Refactoring
- [x] Internal code
- [ ] External Code / Libraries
- [ ] Documentation
- [ ] Other: \_____

Closes : #266 

## Description
Migrate gradle to kotlin as it is more secure

See [this](https://docs.gradle.org/current/userguide/migrating_from_groovy_to_kotlin_dsl.html) for more info on how to migrate

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -0,0 +1 @@
+rootProject.name = "BlackOnion-Bot"
```
Comment (SIMULATAN) [inline settings.gradle.kts:1]: Can we somehow eliminate this file all together? Is it even required?

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+			eachFile {
+				logger.info("  -> Copying ${it.name}...")
+			}
+		}
+		logger.info("  ---    Dependencies copied.  ---")
+		logger.info("===== Dependencies downloaded ======")
+	}
+}
+
+val getLoc () {
```
Comment (SIMULATAN) [inline build.gradle.kts:201]: Please remove the space between the method name and brackets

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+		}
+
+configurations { all { exclude(group = "org.slf4j", module = "slf4j-log4j12") } }
+
+configurations {
+	testIntegrationImplementation.extendsFrom implementation
+			testIntegrationRuntimeOnly.extendsFrom runtimeOnly
+
+			testSharedImplementation.extendsFrom implementation
+			testSharedRuntimeOnly.extendsFrom runtimeOnly
```
Comment (SIMULATAN) [inline build.gradle.kts:147]: Too many tab characters

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+		tasks.jacocoTestReport {
+			group = "Reporting"
+			description = "Generate Jacoco coverage reports after running tests."
+			reports {
+				xml.required.set(true)
+				html.required.set(true)
+			}
+			finalizedBy("jacocoTestCoverageVerification")
+		}
```
Comment (SIMULATAN) [inline build.gradle.kts:138]: Too many tab characters

File: build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+task testIntegration (type: Test) {
+	description = 'Runs integration tests.'
+	group = 'verification'
+
+	testClassesDirs = sourceSets.testIntegration.output.classesDirs
+	classpath = sourceSets.testIntegration.runtimeClasspath
+	shouldRunAfter test
+}
+
+test.dependsOn testIntegration // integration tests are part of the default test task
```
Comment (SIMULATAN) [inline build.gradle.kts:128]: :+1: Thanks for adding these comments!

File: build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+dependencies {
+	val testImplementation ( = {
+	testImplementation(
+		it
+				testIntegrationImplementation (it)
+				testSharedImplementation (it)
+	)
+}
```
Comment (SIMULATAN) [inline build.gradle.kts:64]: Tabs messed up again

File: build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,217 @@
+// intellij + gradle + groovy = pain
```
Comment (SIMULATAN) [inline build.gradle.kts:1]: is it better with kotlin? if yes, i think you can remove this comment ;)

File: build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+//file:noinspection GroovyUnusedAssignment
+//file:noinspection GrUnresolvedAccess
+plugins {
+	java
+	`java-library`
```
Comment (SIMULATAN) [inline build.gradle.kts:7]: why is this here now

File: build.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,217 @@
+// intellij + gradle + groovy = pain
+
+//file:noinspection GroovyUnusedAssignment
```
Comment (SIMULATAN) [inline build.gradle.kts:3]: unused warning

**Review by SIMULATAN — COMMENTED** [review]
Thanks a lot for doing this 😁 🚀 
EDIT: forgot to say that, you can probably ignore most of these comments seeing as the PR is WIP

File: build.gradle.kts (hunk 8 above)
Comment (RealYusufIsmail) [reply build.gradle.kts]: added by mistake

File: build.gradle.kts [hunk 10]
```diff
@@ -197,20 +192,24 @@ tasks.register("downloadDependencies") {
...
-	int filesCount = 0
-	project.sourceSets.main.allSource.srcDirs.each {
-		File dir ->
-		if (dir.isDirectory()) {
-			dir.eachFileRecurse {
-				File file ->
-				if (file.isFile()) {
-					file.eachLine(e -> linesOfCode++)
-					filesCount++
+tasks.register("getLoc") {
```
Comment (SIMULATAN) [inline build.gradle.kts]: I don't know if this is right, but this shouldn't be a task (I think), but rather a helper method that is used in the build task

Comment (RealYusufIsmail) [reply build.gradle.kts]: ignore for know
